Optimized recycling code and removed reduntant entries.

Nimryel opened the pull request:
There was a bunch of recycle code that did nothing. Improved the already existing optimized recycling of pokeballs, potions and revives as well as added berries to it. I will add incense to it as well in the future.
Tested & working with dev. branch.

Nimryel commented:
Also added potion count to console, although not sure if it is really necessary; perhaps berry count is more relevant.

DurtyFree commented:
I need opinions on that because I actually thought there were ideas for implementing setting the individual amount of items you would keep, actually not just per type

DurtyFree commented on PoGo.PokeMobBot.Logic/Tasks/RecycleItemsTask.cs:
```diff
@@ -235,57 +154,42 @@ public class RecycleItemsTask
             }
         }
+
+        private static async Task removeItems(int itemCount, ItemId item, CancellationToken cancellationToken, ISession session)
```
Please start functions uppercase:)

littlemonster1 commented:
wasn't berries recycling fixed in merged PR #255 ?

besides that I am all for the head categories instead of individuals

Nimryel commented:
@DurtyFree Fixed. Realistically, there is no need to do that for pokeballs, potions and berries since they are strictly upgrades (no reason to keep the weaker ones over the stronger ones). Perhaps Incense will be different however when the rest of them are released.

Nimryel commented:
@littlemonster1 That code didn't have any actual RecycleItem calls, so it did nothing.

DurtyFree commented:
@Nimryel You are right for full botting this isn't relevant. But for people just partly botting this might be relevant ( to actually be able configuring it properly )

Nimryel commented:
@DurtyFree You have a point. It comes down to opinion I guess; personally, I feel that the config is getting cluttered enough as is. Perhaps there should be a separate ItemsConfig.json where individual values can be optionally set, but by default the automatic method is used?

littlemonster1 commented:
I think even for people partly botting this is not relevant (I am partly botting if at all, just being here since it fascinates me being involved in coding on a fun project for a little)

the arguments presented in #188 are mainly, it currently does not work and before it worked while I could choose. I do agree with Nimryel, there is no reason to keep weaker items unless the algo does not work. No part time botter will be mad to throw a great ball instead of a pokeball as long as he has some type of pokeballs and not 300 berries :).

Nimryel commented:
It's a very simple algorithm, it just always discards the weakest items first until the defined TotalAmountOfXToKeep is reached, I just cleaned it up a bit and added berries. The deleted stuff is from the old way of doing it when the item counts were individually set in the ItemRecycleFilter (which does nothing now).

DurtyFree commented:
Okay guys, you won:) @Nimryel And also @littlemonster1 I love the reason that your just here because it fascinates you. This also drives me in this project:)

@Nimryel We have already planned on better organizing the configurations and split them in different configs.
Will merge that
